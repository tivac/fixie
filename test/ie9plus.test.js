"use strict";

var simple = require("./util.js").simple,
    nested = require("./util.js").nested;

describe(":ie9plus", () => {
    it("should transform :ie9plus(...) into an ie9plus hack (single selector)", () =>
        simple(`
            :ie9plus(.fooga) {
                color: red;
            }
        `)
    );

    it("should transform :ie9plus(...) into an ie9plus hack (multiple selectors)", () =>
        simple(`
            :ie9plus(.fooga .booga) {
                color: red;
            }
        `)
    );

    it("should work with postcss-nested", () =>
        nested(`
            .fooga {
                color: red;
                
                :ie9plus(&) {
                    color: blue;
                }
            }
        `)
    );
});
