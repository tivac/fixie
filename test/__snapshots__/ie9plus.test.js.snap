// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`:ie9plus should transform :ie9plus(...) into an ie9plus hack (multiple selectors) 1`] = `
"@media screen and (min-width:0\\\\0) and (min-resolution: +72dpi) {.fooga .booga {
    color: red;
}
}"
`;

exports[`:ie9plus should transform :ie9plus(...) into an ie9plus hack (single selector) 1`] = `
"@media screen and (min-width:0\\\\0) and (min-resolution: +72dpi) {.fooga {
    color: red;
}
}"
`;

exports[`:ie9plus should work with postcss-nested 1`] = `
".fooga {
    color: red;
}
    
    @media screen and (min-width:0\\\\0) and (min-resolution: +72dpi) {
    
    .fooga {
        color: blue;
    }
}"
`;
